Rafiq

rrhabib opened the pull request:
Big update on the process of displaying the movie to be rated:
1. Changed the associated JavaScript function and Python endpoint from Get to Post for added security and standardization with our other endpoints.
2. Converted the movie_option() endpoint output from a list to a dictionary to make it easier to reference and understand
3. Created a universal variable for the movie rating process so that we can properly track which movie the user is currently rating
4. Modified some sections of the rating.html page so that it worked properly with the returnMovie() JavaScript function

With these changes, the "display movie to be rated" story is complete, save for final testing once this code has been deployed to the server.

rrhabib committed:
reworked the logic for obtaining and displaying movies to be rated by the user

rrhabib committed:
structured changes for get/post changes in Javascript and Python

rrhabib committed:
fixed conflict between main and branch

rrhabib committed:
begain conversion from Get requests to Put requests

rrhabib committed:
completed local testing of conversion of endpoints from Get to Post and reformatted the full stack connnection for dipslaying movie ratings for better form and more clarity

rrhabib committed:
Consolidated various changes in the ratings.html file based on recent JavaScript changes

AdamSyed commented on application.py:
```diff
@@ -355,5 +372,6 @@ def movie_option(userID):
 if __name__ == '__main__':
-    # application.run(host='0.0.0.0')
+
+    #application.run(host='0.0.0.0')
```
I believe we should not comment this out as the server needs this line to run. Since we want to have the master branch represent the latest shippable code we should uncomment this.

rrhabib replied:
Thanks for catching this.  I have committed the changes you have suggested. Should be good to deploy now.

AdamSyed commented on application.py:
```diff
@@ -295,31 +296,47 @@ def all_movie_ratings():
+        movieData = {'userID':userID, 'movieID':'N/A', 'movie_name':'N/A', 'movie_genre':'N/A', 'movie_director':'N/A'}
+    
+    #finally, we return the results
+    return jsonify(movieData)
```
@Great commenting in this endpoint. Nice work Rafiq!  💯

rrhabib replied:
Thanks!

AdamSyed commented on application.py:
```diff
@@ -1,6 +1,7 @@
 from flask import Flask, request, jsonify
 from flask_sqlalchemy import SQLAlchemy
 from flask_marshmallow import Marshmallow
+import numpy as np
```
We may need to add this to requirements.txt since this is a new library.

rrhabib replied:
Yes.  We can test this when we deploy.  I didn't have to install the library explicitly on my local pc so I'm thinking that it may not be needed on requirements.txt

AdamSyed commented on rating.html:
```diff
@@ -47,7 +47,8 @@
 </head>
- <body>
+
+ <body onload="returnMovie()">>
```
You may want to consider specifying the "onload" trigger in JavaScript. To do this we would setup separate JS files for each page, which may also help with merge conflicts. What are your thoughts?

rrhabib replied:
I'm a bit confused as to how this would work so we should have a call to go over it.  Regardless, lets wait until the next sprint to split the JS file because that will be confusing for the team this late in this current sprint.

AdamSyed reviewed (commented):
Great work Rafiq! I appreciate all the comments. I made a few comments of potential improvements and fixes. Take a look and let me know your thoughts.

rrhabib committed:
Update application.py

Uncommented run line for the live server deployment and commented the local testing line (as per Adam's comments)

AdamSyed commented:
Nice work!
